pkg/command/context.go

```diff
@@ -620,6 +638,38 @@ func globalLogin(c *configtypes.Context) (err error) {
 }
 
 func globalTanzuLogin(c *configtypes.Context, generateContextNameFunc func(orgName, endpoint string, isStaging bool) string) error {
+	if c.AdditionalMetadata[config.TanzuIdpTypeKey] == config.CSPIdpType {
+		return globalTanzuLoginCSP(c, generateContextNameFunc)
+	} else if c.AdditionalMetadata[config.TanzuIdpTypeKey] == config.UAAIdpType {
+		return globalTanzuLoginUAA(c, generateContextNameFunc)
+	}
+	return errors.New(invalidIdpType)
+}
+
+func globalTanzuLoginUAA(c *configtypes.Context, generateContextNameFunc func(orgName, endpoint string, isStaging bool) string) error {
+	uaaEndpoint := c.AdditionalMetadata[config.TanzuAuthEndpointKey].(string)
+	log.V(7).Infof("Login to UAA endpoint: %s", uaaEndpoint)
+
+	claims, err := doInteractiveLoginAndUpdateContext(c, uaaEndpoint)
+	if err != nil {
+		return err
+	}
+
+	// UAA-based authentication does not provide org id or name yet
+	orgName := ""
```

Comment:
OrgName which gets configured in the `context` as part of additionalMetadata is used by all the plugins to show log messages. Keeping this empty might make the log message incomplete.

Reply:
We can do this as part of follow up as well.